src/net/RTP/RTPSession.cs

```diff
@@ -609,6 +633,36 @@ public virtual SetDescriptionResultEnum SetRemoteDescription(SdpType sdpType, SD
                     var remoteTrack = new MediaStreamTrack(announcement.Media, true, announcement.MediaFormats.Values.ToList(), mediaStreamStatus, announcement.SsrcAttributes);
                     addTrack(remoteTrack);
 
+                    if (IsSecure)
+                    {
+                        if (announcement.Transport != RTP_SECUREMEDIA_PROFILE)
+                        {
+                            logger.LogError($"Error negotiating secure media. Invalid Transport {announcement.Transport}.");
+                            return SetDescriptionResultEnum.CryptoNegotiationFailed;
+                        }
+                        
+                        if (announcement.SecurityDescriptions.Count(s => SrtpCryptoSuites.Contains(s.CryptoSuite)) == 0)
+                        {
+                            logger.LogError("Error negotiating secure media. No compatible crypto suite.");
+                            return SetDescriptionResultEnum.CryptoNegotiationFailed;
+                        }
+
+                        if (!m_srtpHandler.SetupRemote(announcement.SecurityDescriptions, sdpType))
+                        {
+                            logger.LogError("Error negotiating secure media. Incompatible crypto parameter.");
+                            return SetDescriptionResultEnum.CryptoNegotiationFailed;
+                        }
+
+                        if (m_srtpHandler.IsNegotiationComplete)
+                        {
+                            SetSecurityContext(
+                                m_srtpHandler.ProtectRTP,
+                                m_srtpHandler.UnprotectRTP,
+                                m_srtpHandler.ProtectRTCP,
+                                m_srtpHandler.UnprotectRTCP);
+                        }
+                    }
+
                     if (announcement.Media == SDPMediaTypesEnum.audio)

```

Comment:
This change breaks WebRTC. It uses SRTP but does not use the same SDP attributes as a VoIP call. If you run the integration tests in Visual Studio you'll see the failures.